### PR TITLE
pcp: Clean up docs, error handling, and dead code 

### PR DESCRIPTION
**Pull request by cjpatton**

### CODE REVIEW

**Review by tgeoghegan — CHANGES_REQUESTED** [review]

File: src/pcp.rs [hunk 1]
```diff
@@ -1,132 +1,63 @@
 // SPDX-License-Identifier: MPL-2.0
 
 //! **(NOTE: This module is experimental. Applications should not use it yet.)** This module
-//! implements a fully linear PCP ("Probabilistically Checkable Proof") system based on
-//! \[[BBC+19](https://eprint.iacr.org/2019/188), Theorem 4.3\].
+//! defines the [`Value`] trait, the main building block of the [`crate::vdaf::prio3`] VDAF.
```
[the remark this answers is not in the document]
Comment (cjpatton) [reply src/pcp.rs:4]: THAT'S what I was looking for! Thanks

File: src/pcp.rs [hunk 2]
```diff
@@ -282,58 +177,15 @@ pub trait Value:
...
-    ///     query(&input_shares[1], &proof_shares[1], &query_rand, &joint_rand).unwrap(),
-    /// ];
-    ///
-    /// let verifier = Verifier::try_from(verifier_shares.as_slice()).unwrap();
-    /// let res = decide(&input_shares[0], &verifier).unwrap();
-    /// assert_eq!(res, true);
-    /// ```
+    //
+    // TODO(cjpatton) Deprecate this method. The proper way to do this should be to normalizing by
+    // dividing by the number of shares.
```
[the remark this answers is not in the document]
Comment (cjpatton) [reply src/pcp.rs:182]: Ack. I'm going to leave as-is to minimize rebase conflicts. (This comment gets deleted later in the PR queue).

File: src/pcp/types.rs [hunk 3]
```diff
@@ -410,20 +420,18 @@ mod tests {
             data: vec![],
             range: poly_range_check(0, 2),
         };
-        assert_matches!(
-            malformed_x.valid(&mut malformed_x.gadget(), &[]),
-            Err(PcpError::CircuitInLen)
-        );
+        malformed_x
+            .valid(&mut malformed_x.gadget(), &[])
+            .unwrap_err();
```
[the remark this answers is not in the document]
Comment (cjpatton) [reply src/pcp/types.rs:425]: Ack, This code changes later in the queue. Alright if I just make a note to fix this up later?

File: src/vdaf.rs [hunk 4]
```diff
@@ -3,7 +3,7 @@
 //! **(NOTE: This module is experimental. Applications should not use it yet.)** This module
 //! implements the prio3 Verifiable Distributed Aggregation Function specified in
 //! [[VDAF](https://cjpatton.github.io/vdaf/draft-patton-cfrg-vdaf.html)]. It is constructed from a
-//! [`Value`].
+//! [`crate::pcp::Value`].
```
Comment (tgeoghegan) [inline src/vdaf.rs:6]: One more nit:
```suggestion
//! [`Value`](crate::pcp::Value).
```

**Review by tgeoghegan — APPROVED** [review]

**Comment by cjpatton** [comment]
Squashed and rebased.

**Pull request merged**